services/QuillLMS/engines/evidence/lib/evidence/evidence/automl_check.rb

```diff
@@ -14,37 +16,73 @@ def initialize(entry, prompt, previous_feedback=[])
-      highlight = feedback.highlights.map do |h|
+      if matched_low_confidence_rule
+        feedback = matched_low_confidence_rule.determine_feedback_from_history(@previous_feedback)
+        highlight = feedback.highlights.map do |h|
+          {
+            type: h.highlight_type,
+            text: h.text,
+            category: ''
+          }
+        end
         {
-          type: h.highlight_type,
-          text: h.text,
-          category: ''
+          feedback: feedback.text,
+          feedback_type: Rule::TYPE_LOW_CONFIDENCE,
+          optimal: matched_low_confidence_rule.optimal,
+          entry: @entry,
+          concept_uid: matched_low_confidence_rule&.concept_uid || '',
+          rule_uid: matched_low_confidence_rule&.uid || '',
+          hint: matched_low_confidence_rule&.hint,
+          highlight: highlight,
+          api: {
+            original_rule_confidence: @confidence_score,
+            original_rule_uid: matched_automl_rule&.uid,
+            original_rule_name: matched_automl_rule&.name
+          }
         }
-      end
-      {
-        feedback: feedback.text,
-        feedback_type: Rule::TYPE_AUTOML,
-        optimal: matched_rule.optimal,
-        entry: @entry,
-        concept_uid: matched_rule&.concept_uid || '',
-        rule_uid: matched_rule&.uid || '',
-        hint: matched_rule&.hint,
-        highlight: highlight,
-        api: {
-          confidence: @confidence_score
+      else
+        feedback = matched_automl_rule.determine_feedback_from_history(@previous_feedback)
+        highlight = feedback.highlights.map do |h|
+          {
+            type: h.highlight_type,
+            text: h.text,
+            category: ''
+          }
+        end
+        {
+          feedback: feedback.text,
+          feedback_type: Rule::TYPE_AUTOML,
+          optimal: matched_automl_rule.optimal,
+          entry: @entry,
+          concept_uid: matched_automl_rule&.concept_uid || '',
+          rule_uid: matched_automl_rule&.uid || '',
+          hint: matched_automl_rule&.hint,
+          highlight: highlight,
+          api: {
+            confidence: @confidence_score
+          }
         }
-      }
+      end
     end
     # rubocop:enable Metrics/CyclomaticComplexity
 
-    private def matched_rule
-      @matched_rule ||= fetch_matched_rule
+    private def matched_low_confidence_rule
+      return unless matched_automl_rule && !matched_automl_rule.optimal && @confidence_score < LOW_CONFIDENCE_THRESHOLD
+
+      @matched_low_confidence_rule ||= fetch_matched_low_confidence_rule
+    end
+
+    private def fetch_matched_low_confidence_rule
+      @prompt.rules&.find_by(rule_type: Evidence::Rule::TYPE_LOW_CONFIDENCE)
+    end
+
+    private def matched_automl_rule
+      @matched_automl_rule ||= fetch_matched_automl_rule
     end
 
-    private def fetch_matched_rule
+    private def fetch_matched_automl_rule
```

Comment:
I think that most of this can actually be consolidated into a `fetch_matched_rule` function that does the work to figure out whether the "matched" rule is of type AutoML or of type LowConfidence.  Something like:
```
private def fetch_matched_automl_rule
  return unless @automl_model

  @google_automl_label, @confidence_score = @automl_model.fetch_automl_label(@entry)
  if @confidence_score < LOW_CONFIDENCE_THRESHOLD
    @prompt.rules&.find_by(rule_type: Evidence::Rule::TYPE_LOW_CONFIDENCE)
  else
    @prompt.rules.joins(:label).find_by(comprehension_labels: {name: @google_automl_label})
end
```
Then you just have a single `matched_rule` value that you can use to build your payload above, and you can figure out which `api` values based on the `matched_rule.rule_type` and set `rule_type` by `matched_rule.rule_type`.

Reply:
Hey @anathomical  - I started trying to implement this and then realized that it is a little complicated by the fact that we do need the `automl` values for the `low_confidence` rule for the api values, so I think it still makes sense to have separate methods that define both. I did refactor the top function in the way you suggested - let me know if you think there's more cleanup we should do here.